### PR TITLE
Mitigate random failures in the MutableProperty async deadlock test and Action infinite feedback test.

### PR DESCRIPTION
**Pull request by andersio**
#### Checklist
- ~~Updated CHANGELOG.md.~~

### CODE REVIEW

**Comment by mdiep** [comment]
It doesn't seem to have helped. 🙈

**Comment by andersio** [comment]
Eww.

File: Tests/ReactiveSwiftTests/PropertySpec.swift [hunk 1]
```diff
@@ -285,7 +285,7 @@ class PropertySpec: QuickSpec {
 
 					target! <~ source
 
-					queue.async(group: group) {
+					queue.async(group: group, flags: .barrier) {
```
Comment (andersio) [inline Tests/ReactiveSwiftTests/PropertySpec.swift]: It limits the concurrency, but IIRC does not affect the assertion here which was about the mitigated data race between the two blocks, not across iterations.

Comment (mdiep) [inline Tests/ReactiveSwiftTests/PropertySpec.swift]: I think it does:

> Instead, the queue waits until its currently executing blocks finish executing. At that point, the barrier block executes by itself. Any blocks submitted after the barrier block are not executed until the barrier block completes.

(https://developer.apple.com/documentation/dispatch/1452797-dispatch_barrier_async?language=objc)

I think you'd need a separate, empty barrier block.

File: Tests/ReactiveSwiftTests/ActionSpec.swift [hunk 2]
```diff
@@ -162,25 +162,15 @@ class ActionSpec: QuickSpec {
...
-
-						disposable += lowPriority.schedule {
-							done()
+							count += 1
+							expect(count) == 1
 						}
-					}
-
-					disposable.dispose()
 				}
```
Comment (mdiep) [inline Tests/ReactiveSwiftTests/ActionSpec.swift]: Was this test also occasionally failing?

Comment (andersio) [inline Tests/ReactiveSwiftTests/ActionSpec.swift]: Yes.

File: Tests/ReactiveSwiftTests/PropertySpec.swift (hunk 1 above)
Comment (andersio) [inline Tests/ReactiveSwiftTests/PropertySpec.swift]: Ugh, you are right.

**Pull request merged**